### PR TITLE
[DO-1730] Create PEP11 scoring rule

### PR DESCRIPTION
**Pull request by Andrii17s**
Royalty exceeds 20% of the total income indicated in the declaration.

### CODE REVIEW

File: business_register/pep_scoring/rules.py [hunk 1]
```diff
@@ -104,3 +104,46 @@ def calculate_weight(self):
...
+    def calculate_weight(self):
+        declarations = Declaration.objects.filter(
+            pep_id=self.pep.id,
+        ).values('id', 'year')[::1]
+        declaration_ids = {}
+
+        for declaration in declarations:
+            year = declaration['year']
+            if not declaration_ids.__contains__(year):
+                declaration_ids[declaration['year']] = list()
```
Comment (roman-tiukh) [inline business_register/pep_scoring/rules.py:125]:
```suggestion
            if year not in declaration_ids:
                declaration_ids[declaration['year']] = []
```

**Review by roman-tiukh — APPROVED** [review]

File: business_register/pep_scoring/rules.py [hunk 2]
```diff
@@ -173,37 +173,29 @@ class IsMuchRoyalty(BaseScoringRule):
...
+
+    class DataSerializer(serializers.Serializer):
+        royalty_UAH = serializers.IntegerField(min_value=0, required=True)
+        assets_UAH = serializers.IntegerField(min_value=0, required=True)
+        year = serializers.IntegerField(min_value=0, required=True)
+
+    def calculate_weight(self) -> tuple[int or float, dict]:
+        assets_UAH = 0
+        royalty_UAH = 0
+        incomes = Income.objects.filter(
```
[the remark this answers is not in the document]
Comment (Andrii17s) [reply business_register/pep_scoring/rules.py:186]: But we need all other Income types to calculate how much is 20%

**Review by IegorSoboliev — APPROVED** [review]

File: business_register/pep_scoring/rules.py [hunk 3]
```diff
@@ -225,6 +225,43 @@ def calculate_weight(self) -> Tuple[Union[int, float], dict]:
         return 0, {}
 
 
+@register_rule
+class IsMuchRoyalty(BaseScoringRule):
```
Comment (criticus) [inline business_register/pep_scoring/rules.py:229]: class name has to be noun

**Review by IegorSoboliev — APPROVED** [review]

File: business_register/pep_scoring/rules.py [hunk 4]
```diff
@@ -225,6 +225,43 @@ def calculate_weight(self) -> Tuple[Union[int, float], dict]:
...
+        royalty_UAH = 0
+        incomes = Income.objects.filter(
+            declaration_id=self.declaration.id,
+        ).values_list('amount', 'type')[::1]
+        for income in incomes:
+            try:
+                assets_UAH += income[0]
+                if income[1] == Income.DIVIDENDS:
+                    royalty_UAH += income[0]
+            except:
```
Comment (roman-tiukh) [inline business_register/pep_scoring/rules.py:253]: except all errors its bad practice

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

**Review by roman-tiukh — APPROVED** [review]

**Pull request merged**